File: Algo.Analytics.FSharp/EmptyAnalyticsScript.fs
namespace StockSharp.Algo.Analytics

open System
open System.Threading
open System.Threading.Tasks

open Ecng.Common
open Ecng.Logging

open StockSharp.Messages
open StockSharp.Algo.Analytics
open StockSharp.Algo.Storages
open StockSharp.Algo.Candles

/// <summary>
/// The empty analytic strategy.
/// </summary>
type EmptyAnalyticsScript() =
    interface IAnalyticsScript with
        member this.Run
            (
                logs: ILogReceiver,
                panel: IAnalyticsPanel,
                securities: SecurityId[],
                from: DateTime,
                ``to``: DateTime,
                storage: IStorageRegistry,
                drive: IMarketDataDrive,
                format: StorageFormats,
                timeFrame: TimeSpan,
                cancellationToken: CancellationToken
            ) : Task =
            
            if securities.Length = 0 then
                logs.LogWarning("No instruments.")
                Task.CompletedTask
            else
                // !! add logic here
                Task.CompletedTask

File: Algo.Analytics.CSharp/EmptyAnalyticsScript.cs
namespace StockSharp.Algo.Analytics;

/// <summary>
/// The empty analytic strategy.
/// </summary>
public class EmptyAnalyticsScript : IAnalyticsScript
{
	Task IAnalyticsScript.Run(ILogReceiver logs, IAnalyticsPanel panel, SecurityId[] securities, DateTime from, DateTime to, IStorageRegistry storage, IMarketDataDrive drive, StorageFormats format, TimeSpan timeFrame, CancellationToken cancellationToken)
	{
		if (securities.Length == 0)
		{
			logs.LogWarning("No instruments.");
			return Task.CompletedTask;
		}

		// !! add logic here

		return Task.CompletedTask;
	}
}